kevinm23nelson/CryptoPal
# Feature/cypress testing

## What does this PR do?
This PR adds Cypress tests to ensure that various user flows in the application are functioning correctly. These tests cover scenarios such as navigating through the app, interacting with UI elements, and verifying data rendering on different pages.

## Where should the reviewer start?
The reviewer should start by looking at the Cypress test files located in the `cypress/integration` directory. Reviewing the test files will provide an understanding of the different user flows being tested and the structure of the tests.

## How should this be manually tested?
To manually test the changes, run the Cypress test suite using the command `npx cypress open` or `npx cypress run`. This will launch Cypress and execute the tests, providing visual feedback and logs for each test scenario. Verify that all tests pass and that the application behaves as expected.

## Any background context you want to provide?
These tests were created to ensure the robustness of the application by simulating real user interactions. The tests cover critical user flows, such as adding and removing favorites, viewing currency details, and verifying that the UI updates correctly based on user actions.

## What are the relevant tickets (if any)?
#13 

## Screenshots (if appropriate)
<img width="1601" alt="Screenshot 2024-08-01 at 2 19 50 PM" src="https://github.com/user-attachments/assets/8c1590a7-4fe3-44c4-bfa6-c98f8d9a563e">


## Questions:
- Are there any additional user flows or edge cases that should be covered by the tests?
- Are the tests comprehensive enough to catch potential regressions in future updates?
- Is the current setup for running and organizing Cypress tests optimal for the team's workflow?
